src/toolchain.rs

```diff
@@ -44,11 +45,23 @@ pub enum UpdateStatus {
 
 impl<'a> Toolchain<'a> {
     pub fn from(cfg: &'a Cfg, name: &str) -> Result<Self> {
+        let v: Vec<&str> = name.split('/').collect();
+        let name = v[0];
+
         let resolved_name = cfg.resolve_toolchain(name)?;
         let path = cfg.toolchains_dir.join(&resolved_name);
+
+        let resolved_name_dated = if v.len() > 1 {
+            let name_dated = name.to_string() + "-" + v[1];
+            cfg.resolve_toolchain(&name_dated)?
```

Comment:
While this will produce a suitable error, it won't be entirely clear that it's due to the `/stuff` part.  I wonder if this needs chaining slightly into a more descriptive error on failure.